js/bybit.js

```diff
@@ -1942,28 +1942,24 @@ module.exports = class bybit extends Exchange {
         ];
     }
 
-    async fetchSpotOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
+    async fetchSpotOHLCV (symbol, timeframe = '1m', since = undefined, limit = 1000, params = {}) {
         await this.loadMarkets ();
         const market = this.market (symbol);
         const request = {
             'symbol': market['id'],
+            'limit': limit, // default is 1000, max is 1000
+            'interval': timeframe,
         };
-        const duration = this.parseTimeframe (timeframe);
-        const now = this.seconds ();
-        let sinceTimestamp = undefined;
-        if (limit === undefined) {
-            limit = 200; // default is 200 when requested with `since`
-        }
-        if (since === undefined) {
-            sinceTimestamp = now - limit * duration;
-        } else {
-            sinceTimestamp = parseInt (since / 1000);
-        }
-        if (limit !== undefined) {
-            request['limit'] = limit; // max 200, default 200
+        if (since !== undefined) {
+            // startTime isn't working, emulated since using endTime
+            // request['startTime'] = since;
+            const duration = this.parseTimeframe (timeframe).toString ();
```

Comment:
@Dan-krm `startTime` isn't working at all? If so, I can try to reach out to them

Reply:
That's a good idea, yeah I just checked again and it doesn't seem to be working at all

Reply:
@Dan-krm so I've checked with them and `startTime` works, it might give the impression that it does not because if you also provide a `limit` it will only retrieve the most recent N entries.

With that said, we should be able to use it here, what do you think?

Reply:
Thanks for doing that, it looks like it's still not working:
```js
async fetchSpotOHLCV (symbol, timeframe = '1m', since = undefined, limit = undefined, params = {}) {
    await this.loadMarkets ();
    const market = this.market (symbol);
    const request = {
        'symbol': market['id'],
        'interval': timeframe,
    };
    if (since !== undefined) {
        request['startTime'] = since;
        // startTime isn't working, emulated since using endTime
        // const duration = this.parseTimeframe (timeframe).toString ();
        // const millisecondDuration = Precise.stringMul (duration, '1000');
        // const timeFromSince = Precise.stringMul (limit.toString (), millisecondDuration);
        // const emulatedSince = Precise.stringAdd (since.toString (), timeFromSince);
        // const emulatedSinceAdjustDurationError = Precise.stringSub (emulatedSince, millisecondDuration);
        // request['endTime'] = this.parseNumber (emulatedSinceAdjustDurationError);
    } else if (limit !== undefined) {
        request['limit'] = limit; // default is 1000, max is 1000. Can't be used at the same time as since
    }
    const response = await this.publicGetSpotV3PublicQuoteKline (this.extend (request, params));
    //
    //     {
    //         "retCode": 0,
    //         "retMsg": "OK",
    //         "result": {
    //         "list": [
    //             {
    //             "t": 1659430380000,
    //             "s": "BTCUSDT",
    //             "sn": "BTCUSDT",
    //             "c": "21170.14",
    //             "h": "21170.14",
    //             "l": "21127.86",
    //             "o": "21127.86",
    //             "v": "0.907276"
    //             }
    //         ]
    //         },
    //         "retExtInfo": {},
    //         "time": 1659430400353
    //     }
    //
    const result = this.safeValue (response, 'result', {});
    const ohlcvs = this.safeValue (result, 'list', []);
    return this.parseOHLCVs (ohlcvs, market, timeframe, since, limit);
}
```
`{ symbol: 'BTCUSDT', interval: '5m', startTime: 1672531200000 }`
```
bybit.fetchOHLCV (BTC/USDT, 5m, 1672531200000)
2023-01-23T20:04:36.871Z iteration 0 passed in 523 ms

1674204300000 | 20945.66 | 20945.77 | 20939.01 | 20940.08 |   4.568522
1674204600000 | 20940.08 | 20940.24 | 20924.41 | 20929.29 |    7.27833
1674204900000 | 20929.29 | 20941.43 | 20927.14 | 20941.43 |   5.103089
...
1674503400000 | 22719.99 | 22742.54 | 22686.03 | 22740.58 |  27.525776
1674503700000 | 22740.58 | 22797.51 | 22740.58 | 22795.65 |  42.110836
1674504000000 | 22795.65 | 22799.12 | 22761.09 | 22779.04 |  35.985264
1000 objects
```

Reply:
@Dan-krm Yeah I'm seeing, I suspect they do this: filter all candles after since and they get a list with, let's say 5k but since they only return 1k and they cut the 1k newest candles from the list, it looks like the limit did not work and yeah if they're doing this it's a terrible approach because it's completely useless unless we also provide an `endTime`

Reply:
@carlosmiei My approach emulating since using endTime seems to work for all timeframes if you think it's okay to merge, Let me know if you don't think this is okay to merge and I can create an example emulating since on the user side to answer the issue that this resolves